packages/flutter/lib/src/rendering/semantics.dart

```diff
@@ -137,6 +129,27 @@ class SemanticsNode extends AbstractNode {
   // FLAGS AND LABELS
   // These are supposed to be set by SemanticAnnotator obtained from getSemanticAnnotators
 
+  final Set<SemanticAction> _actions = new Set<SemanticAction>();
+
+  void addAction(SemanticAction action) {
+    if (_actions.add(action))
+      _markDirty();
+  }
+
+  void addHorizontalScrollingActions() {
+    addAction(SemanticAction.scrollLeft);
+    addAction(SemanticAction.scrollRight);
+  }
+
+  void addVerticalScrollingActions() {
+    addAction(SemanticAction.scrollUp);
+    addAction(SemanticAction.scrollDown);
+  }
+
```

Comment:
Probably a good idea.  I was going to wait to add that until I added the first client (which will be Slider).